concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
+        raise ValidationError({
+            'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.CannotComputeTask().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredCannotComputeTask': 'StoredCannotComputeTask model has not same fields as golem_messages.message.tasks.CannotComputeTask()'
+            })
+
+    class CannotComputeTaskReason(ChoiceEnum):
+        WrongCTD = 'WrongCTD'
+        WrongKey = 'WrongKey'
+        WrongAddress = 'WrongAddress'
+        WrongEnvironment = 'WrongEnvironment'
+        NoSourceCode = 'NoSourceCode'
+        WrongDockerImages = 'WrongDockerImages'
+        ConcentRequired = 'ConcentRequired'
+        ConcentDisabled = 'ConcentDisabled'
+
+    if not len(CannotComputeTaskReason.__members__) == len(message.tasks.CannotComputeTask.REASON.__members__):
+        raise ValidationError({
+            'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+        })
+
+    for model_item, golem_message_item in zip(CannotComputeTaskReason.__members__, message.tasks.CannotComputeTask.REASON.__members__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'CannotComputeTaskReason': 'CannotComputeTaskReason Enum has not same fields as golem_messages.message.tasks.CannotComputeTask.REASON'
+            })
+
+    task_to_compute = ForeignKey(StoredTaskToCompute)
+    reason = CharField(max_length=32, choices=CannotComputeTaskReason.choices())
+
+
+class StoredReportComputedTask(AbstractMessageModel):
+
+    ITEMS = [
+        'result_type',
+        'node_name',
+        'address',
+        'node_info',
+        'port',
+        'key_id',
+        'extra_data',
+        'eth_account',
+        'task_to_compute',
+        'size',
+        'package_hash',
+        'multihash',
+        'secret',
+        'options',
+        'header',
+        'sig',
+    ]
+
+    if not len(ITEMS) == len(message.tasks.ReportComputedTask().__slots__):
+        raise ValidationError({
+            'StoredReportComputedTask': 'StoredReportComputedTask model has not same fields as golem_messages.message.tasks.ReportComputedTask()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.ReportComputedTask().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredReportComputedTask': 'StoredReportComputedTask model has not same fields as golem_messages.message.tasks.ReportComputedTask()'
+            })
+
+    result_type = PositiveSmallIntegerField()
+    node_name = CharField(max_length=32)
+    address = CharField(max_length=15)
```

Comment:
There should be constants for these two values. Where do they come from?

Reply:
15 is maximum length of IPv4 address -> 192.168.128.100 -> This gives exactly 15 chars. I'll add it to constants. Node name is not precised yet